packages/astro/src/core/build/generate.ts

```diff
@@ -185,14 +135,15 @@ export async function generatePages(options: StaticBuildOptions, internals: Buil
 					});
 				}
 
-				const ssrEntryURLPage = createEntryURL(filePath, outFolder);
-				const ssrEntryPage = await import(ssrEntryURLPage.toString());
+				const ssrEntryPage = await pipeline.retrieveSsrEntry(pageData.route, filePath);
 				if (options.settings.adapter?.adapterFeatures?.functionPerRoute) {
 					// forcing to use undefined, so we fail in an expected way if the module is not even there.
+					// @ts-expect-error When building for `functionPerRoute`, the module exports a `pageModule` function instead
```

Comment:
I'm a little unclear on what causes this type error. Is it because of the experimental flag?

Reply:
It's because I moved the code inside the pipeline, so now we always use the same method to load the built page. However, when using `functionPerRoute`, the exported data of that entry point is different, and I couldn't find a clever to make type safe.